### ISSUE TITLE
508-defect-3:  Error message needs to tell the user why the email address is invalid

### ISSUE THREAD

**Issue by jeana-adhoc** [department-of-veterans-affairs/va.gov-team#56733]
### Point of contact

@jeana-adhoc

### Severity level

3, Moderate. Should be fixed in 1-3 sprints post-launch.

### Details

The 508 Office Accessibility Audit identified one issue that needs to be fixed.

> On the page, Contact Information 2,  When doing validation on a field and putting in an error message, then the error message must be described to the user.  Simply saying invalid text is not good enough, you need to tell the user why it is invalid.  The email error just says “Please enter a valid email address” it needs to tell the user why the email address is invalid.

508 Office Report: [Digitized form 26-4555 Ad pated Housing Grant 508 Audit Report 20230411 CEW.docx](https://github.com/department-of-veterans-affairs/va.gov-team/files/11212208/Digitized.form.26-4555.Ad.pated.Housing.Grant.508.Audit.Report.20230411.CEW.docx)

### Reproduction steps

1. Navigate to the "[Your Contact information](https://www.va.gov/housing-assistance/disability-housing-grants/apply-for-grant-form-26-4555/contact-information-2)" page on the 26-4555 form
2. Type in an invalid email address
3. The error message reads "Please enter a valid email address"

<img width="528" alt="image" src="https://user-images.githubusercontent.com/117098918/231470972-c1aa1c39-e792-4bcb-aa09-d5a3e44bd24b.png">

### Proposed solution or next steps

While the current implementation follows the guidance in [design system](https://design.va.gov/patterns/ask-users-for/email-address) we need to have more specific language in the error message.

When there is no "@" in the email address, the error message, approved by the sitewide content team, should read:

> Enter a valid email address. Your address can only have letters, numbers, the @ symbol, and a period, with no spaces.

This change should *only* impact our forms, and should not edit the default message in the component itself at this time.

### References, articles, or WCAG support

1. [WCAG 3.3.3. Error suggestions](https://www.w3.org/TR/UNDERSTANDING-WCAG20/minimize-error-suggestions.html#:~:text=Specific%20Benefits%20of%20Success%20Criterion,and%20how%20to%20correct%20it.)

### Type of issue

- [X] Screenreader
- [ ] Keyboard
- [ ] Focus
- [ ] Headings
- [ ] Color
- [ ] Zoom
- [ ] Semantics
- [ ] Axe-core
- [X] Something else

**Comment by jeana-adhoc** [comment]
In talking with @DanielleThierryUSDSVA she asked that we not change the default message just yet. They're having a retreat soon where all of the error messages are being evaluated.
